eclair-core/src/main/scala/fr/acinq/eclair/io/Peer.scala

```diff
@@ -28,109 +29,91 @@ import fr.acinq.eclair.Features.Wumbo
 import fr.acinq.eclair.Logs.LogCategory
 import fr.acinq.eclair.blockchain.EclairWallet
 import fr.acinq.eclair.channel._
-import fr.acinq.eclair.crypto.TransportHandler
-import fr.acinq.eclair.router._
 import fr.acinq.eclair.wire._
 import fr.acinq.eclair.{wire, _}
 import kamon.Kamon
-import scodec.Attempt
-import scodec.bits.{BitVector, ByteVector}
-
-import scala.compat.Platform
-import scala.concurrent.duration._
-import scala.util.Random
+import scodec.bits.ByteVector
 
 /**
+ * This actor represents a logical peer. There is one [[Peer]] per unique remote node id at all time.
+ *
+ * The [[Peer]] actor is mostly in charge of managing channels, stays alive when the peer is disconnected, and only dies
+ * when there are no more channels.
+ *
+ * Everytime a new connection is established, it is sent to the [[Peer]] and replaces the previous one.
+ *
  * Created by PM on 26/08/2016.
  */
-class Peer(val nodeParams: NodeParams, remoteNodeId: PublicKey, authenticator: ActorRef, watcher: ActorRef, router: ActorRef, relayer: ActorRef, paymentHandler: ActorRef, wallet: EclairWallet) extends FSMDiagnosticActorLogging[Peer.State, Peer.Data] {
+class Peer(val nodeParams: NodeParams, remoteNodeId: PublicKey, switchboard: ActorRef, router: ActorRef, watcher: ActorRef, relayer: ActorRef, paymentHandler: ActorRef, wallet: EclairWallet) extends FSMDiagnosticActorLogging[Peer.State, Peer.Data] {
 
   import Peer._
 
-  startWith(INSTANTIATING, Nothing())
+  val reconnectionTask: ActorRef = context.actorOf(ReconnectionTask.props(nodeParams, remoteNodeId, switchboard, router), "reconnection-task")
```

Comment:
We're not watching the `reconnectionTask`, so what happens if it crashes? Shouldn't we restart it?

Reply:
Restarting it won't be enough as we are making assumptions on the initial case. I think not reconnecting is relatively harmless and I would just let it stop. We would still be reachable but would not attempt to make connections.